### PR TITLE
Compatibility with `pyscfad`

### PR DESCRIPTION
**Pull request by obackhouse**
Currently incredibly hacky support for `pyscfad` with `jax` backends. Will take a lot of refactoring to properly support.

### CODE REVIEW

**Comment by obackhouse** [comment]
@mkrompiec here is an example script (copied from a pyscfad example) comparing RCCSD to the pyscfad implementation and then doing the same thing with UCCSD:

```
import jax
import sys
import os
os.environ["EBCC_BACKEND"] = "jax"
from pyscfad import gto, scf, cc, config
from ebcc import EBCC, default_log
from ebcc.util import Namespace

config.update("pyscfad_scf_implicit_diff", True)
config.update("pyscfad_ccsd_implicit_diff", True)

mol = gto.Mole(
    atom="H 0 0 0; F 0 0 1",
    basis="cc-pvdz",
    verbose=0,
)
mol.build()

def ccsd(mol, dm0=None):
    mf = scf.RHF(mol)
    mf.kernel(dm0)
    ccsd = cc.RCCSD(mf)
    ccsd.kernel()
    return ccsd.e_tot

jac = jax.jacrev(ccsd)(mol)
print("*** PySCF-AD ***")
print(f"Nuclear gradient:\n{jac.coords}")
print(f"Gradient wrt basis exponents:\n{jac.exp}")
print(f"Gradient wrt basis contraction coefficients:\n{jac.ctr_coeff}")

def ccsd(mol, dm0=None):
    mf = scf.RHF(mol)
    mf.kernel(dm0)
    ccsd = EBCC(mf, ansatz="CCSD")
    ccsd.kernel()
    return ccsd.e_tot

jac = jax.jacrev(ccsd)(mol)
print("*** PySCF-AD + ebcc ***")
print(f"Nuclear gradient:\n{jac.coords}")
print(f"Gradient wrt basis exponents:\n{jac.exp}")
print(f"Gradient wrt basis contraction coefficients:\n{jac.ctr_coeff}")

def uccsd(mol, dm0=None):
    mf = scf.UHF(mol)
    mf.kernel(dm0)
    ccsd = EBCC(mf, ansatz="CCSD")
    ccsd.kernel()
    return ccsd.e_tot

jac = jax.jacrev(uccsd)(mol)
print("*** PySCF-AD + ebcc (UHF) ***")
print(f"Nuclear gradient:\n{jac.coords}")
print(f"Gradient wrt basis exponents:\n{jac.exp}")
print(f"Gradient wrt basis contraction coefficients:\n{jac.ctr_coeff}")
```

The gradients are in good agreement with pyscfad for the restricted case.

Note that I default to 64 bit jax:
https://github.com/BoothGroup/ebcc/blob/b687c618d583d4db1cd44457522756aa3b370e16/ebcc/backend/_jax.py#L8

Let me know if you have any thoughts, if this is useful, if you need any other functionality etc.

**Comment by obackhouse** [comment]
This can get merged, but I'll wait on https://github.com/fishjojo/pyscfad/issues/60 to be fixed rather than using the github link for the installation.

**Comment by obackhouse** [comment]
> This can get merged, but I'll wait on [fishjojo/pyscfad#60](https://github.com/fishjojo/pyscfad/issues/60) to be fixed rather than using the github link for the installation.

nvm my bad, just need to hack in the imports. I fixed the logging now and I'm happy to merge, it's not completely covered but I don't require that for non-numpy backends
